### ISSUE TITLE
Automated swearing filter

### ISSUE THREAD

**Issue by fluxionary** [obl3pplifp/BlS-issue-tracker#200]
aleksbuilder writes: 
```
Naturally, admins cannot be online all the time, but sometimes 
players come to the server who start swearing (some of them 
specially come for this). Today, for example, there were 2 such 
players who swore. Could give some old players the privilege 
of depriving such players of a "voice" in the chat? Or make a 
script that will automatically rob them of their “voice” for several 
hours if they swear 2 times in 5 minutes, for example ?
```

**Comment by fluxionary** [comment]
The difference between this and previous swearing filter ideas, is the idea of just robbing them of their "voice" privs for a period of time.

**Comment by ragulanramkumar** [comment]
a /votemute would be a good idea, but this assumes most players care (I
imagine most would only do something if it stopping them from reading the
chat) and will be around when they're online, which isn't always the case

**Comment by WulftheNordicRanger** [comment]
My issue with /votemute and /votekick are that it could be used to gang up on and bully other players

**Comment by fluxionary** [comment]
hm. i don't like the idea of voting people off the server, because it can so easily be abused, but it *might* make sense in conjunction w/ a "bad language" filter; you could only be voted off if the server had flagged you.

**Comment by Lanhild** [comment]
Solved :)

I made my mod swearing filter based on some anti spam, here's the link: https://github.com/Lanhild/filter-master

Normally fully functionnal.

**Comment by TechDudie** [comment]
Yes, lanhildy, your mod is great, but there’s a problem, there maybe a typo in the filtered word and then your mod won’t catch it. I have some experience with this (I did write some algorithms similar to this at one point), but I have a problem, I (as you can tell from my picture) program in Python, so you can convert the code from python to lua or write a program to communicate between the two programs.

**Comment by TechDudie** [comment]
I have a repo that fixes this issue, but it’s in python, and I don’t know Lua. https://github.com/TechDudie/filter I probably should mention that I never tested it yet. I will when I get the chance.

**Comment by Lanhild** [comment]
Probably better to use mine, since it's in Lua, and Lua is easier to fix bugs,that's what I think

**Comment by TechDudie** [comment]
Well why not translate it from python to lua

**Comment by TechDudie** [comment]
You can write some sort of python and lua signal transfer, I’ll try to help even if I don’t know lua, and python in my opinion is better to implement typo algorithms

**Comment by TechDudie** [comment]
I have lots of time off (on the weekends) enough for me to mantain 3 repos

**Comment by fluxionary** [comment]
Ixnay on any python in the mods. 

Has this been tested w/ the rest of the BlS mods, verbana in particular? We're already doing a bunch of weird stuff to chat functions.

**Comment by Lanhild** [comment]
If you are talking about this: https://github.com/Lanhild/filter-master

Yes I tested it with a copy of BlS mods
And the results were good
